components/Filter/FilterBy/Trait.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Checkbox,
+  CheckboxGroup,
+  Flex,
+  Heading,
+  SkeletonText,
+  Stack,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import useTranslation from 'next-translate/useTranslation'
+import { FC, useCallback, useMemo } from 'react'
+import { UseFormReturn } from 'react-hook-form'
+import { useFetchCollectionTraitsQuery } from '../../../graphql'
+import { Filter } from '../../../hooks/useAssetFilterFromQuery'
+
+type Props = {
+  collection:
+    | {
+        chainId: number
+        address: string
+      }
+    | undefined
+  filter: Filter
+  formValues: UseFormReturn<Filter, any, undefined>
+  propertySearch: string
+  onFilterChange: (filter: Filter) => void
+}
+
+const FilterByTrait: FC<Props> = ({
+  collection,
+  filter,
+  formValues: { watch },
+  propertySearch,
+  onFilterChange,
+}) => {
+  const { t } = useTranslation('components')
+
+  const filterResult = watch()
+
+  const { data } = useFetchCollectionTraitsQuery({
+    variables: {
+      address: (collection && collection.address) || '',
+      chainId: (collection && collection.chainId) || 0,
+    },
+    skip: !collection,
+    ssr: false,
+  })
+
+  const traitsData = data?.collection?.traitsOfCollection.nodes
+
+  const traits = useMemo(() => {
+    if (!traitsData) return
+    if (!propertySearch) return traitsData
+    return traitsData.filter(({ type }) =>
+      type.toLowerCase().includes(propertySearch.toLowerCase()),
+    )
+  }, [propertySearch, traitsData])
+
+  const addTrait = useCallback(
+    (type: string, value: string) => {
+      const existingValues =
+        filterResult.traits.find((x) => x.type === type)?.values || []
+      onFilterChange({
+        ...filterResult,
+        traits: [
+          ...filterResult.traits.filter((x) => x.type !== type),
+          {
+            type,
+            values: [
+              ...(existingValues.filter((x) => x !== value) || []),
+              value,
+            ],
+          },
+        ],
+      })
+    },
+    [onFilterChange, filterResult],
+  )
+
+  const removeTrait = useCallback(
+    (type: string, value: string) => {
+      const existingValues =
+        filterResult.traits.find((x) => x.type === type)?.values || []
+      const traits = filterResult.traits.filter((x) => x.type !== type)
+      if (existingValues.includes(value) && existingValues.length === 1) {
+        onFilterChange({ ...filterResult, traits })
+      } else {
+        onFilterChange({
+          ...filterResult,
+          traits: [
+            ...traits,
+            {
+              type,
+              values: existingValues.filter((x) => x !== value),
+            },
+          ],
+        })
+      }
+    },
+    [onFilterChange, filterResult],
+  )
+
+  if (!collection) return null
+  return !traits ? (
+    new Array(4).fill(0).map((_, index) => (
```

Comment:
The limit is 50 in the query but that will be too much, maybe 10 will be a good middle ground?
```suggestion
    new Array(10).fill(0).map((_, index) => (
```

Reply:
[Change empty state array lengths](https://github.com/liteflow-labs/starter-kit/pull/375/commits/2276be0a80a95fc5c75325b609511557271e6333)